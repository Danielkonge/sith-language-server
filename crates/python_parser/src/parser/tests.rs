mod context;
mod function;
mod invalid_assignments;
mod parser;
mod suite;
