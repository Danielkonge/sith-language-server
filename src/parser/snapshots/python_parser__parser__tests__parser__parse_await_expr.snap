---
source: src/parser/tests.rs
expression: "parse(\"\nawait x\nawait x + 1\nawait a and b\nawait f()\nawait [1, 2]\nawait {3, 4}\nawait {i: 5}\nawait 7, 8\nawait (9, 10)\nawait 1 == 1\n\")"
---
ParsedFile {
    ast: Module {
        body: [
            Expression(
                ExpressionStmt {
                    value: Await(
                        AwaitExpr {
                            value: Id(
                                Identifier {
                                    id: "x",
                                    range: 7..8,
                                },
                            ),
                            range: 1..8,
                        },
                    ),
                    range: 1..8,
                },
            ),
            Expression(
                ExpressionStmt {
                    value: Await(
                        AwaitExpr {
                            value: BinaryOp(
                                BinaryOpExpr {
                                    left: Id(
                                        Identifier {
                                            id: "x",
                                            range: 15..16,
                                        },
                                    ),
                                    op: Add,
                                    right: Literal(
                                        LiteralExpr {
                                            value: Int(
                                                "1",
                                            ),
                                            range: 19..20,
                                        },
                                    ),
                                    range: 15..20,
                                },
                            ),
                            range: 9..20,
                        },
                    ),
                    range: 9..20,
                },
            ),
            Expression(
                ExpressionStmt {
                    value: Await(
                        AwaitExpr {
                            value: BoolOp(
                                BoolOpExpr {
                                    op: And,
                                    values: [
                                        Id(
                                            Identifier {
                                                id: "a",
                                                range: 27..28,
                                            },
                                        ),
                                        Id(
                                            Identifier {
                                                id: "b",
                                                range: 33..34,
                                            },
                                        ),
                                    ],
                                    range: 27..34,
                                },
                            ),
                            range: 21..34,
                        },
                    ),
                    range: 21..34,
                },
            ),
            Expression(
                ExpressionStmt {
                    value: Await(
                        AwaitExpr {
                            value: Call(
                                CallExpr {
                                    lhs: Id(
                                        Identifier {
                                            id: "f",
                                            range: 41..42,
                                        },
                                    ),
                                    args: Arguments {
                                        args: [],
                                        kw_args: [],
                                        range: 42..44,
                                    },
                                    range: 41..44,
                                },
                            ),
                            range: 35..44,
                        },
                    ),
                    range: 35..44,
                },
            ),
            Expression(
                ExpressionStmt {
                    value: Await(
                        AwaitExpr {
                            value: List(
                                ListExpr {
                                    elements: [
                                        Literal(
                                            LiteralExpr {
                                                value: Int(
                                                    "1",
                                                ),
                                                range: 52..53,
                                            },
                                        ),
                                        Literal(
                                            LiteralExpr {
                                                value: Int(
                                                    "2",
                                                ),
                                                range: 55..56,
                                            },
                                        ),
                                    ],
                                    range: 51..57,
                                },
                            ),
                            range: 45..57,
                        },
                    ),
                    range: 45..57,
                },
            ),
            Expression(
                ExpressionStmt {
                    value: Await(
                        AwaitExpr {
                            value: Set(
                                SetExpr {
                                    elements: [
                                        Literal(
                                            LiteralExpr {
                                                value: Int(
                                                    "3",
                                                ),
                                                range: 65..66,
                                            },
                                        ),
                                        Literal(
                                            LiteralExpr {
                                                value: Int(
                                                    "4",
                                                ),
                                                range: 68..69,
                                            },
                                        ),
                                    ],
                                    range: 64..70,
                                },
                            ),
                            range: 58..70,
                        },
                    ),
                    range: 58..70,
                },
            ),
            Expression(
                ExpressionStmt {
                    value: Await(
                        AwaitExpr {
                            value: Dict(
                                DictExpr {
                                    keys: [
                                        Some(
                                            Id(
                                                Identifier {
                                                    id: "i",
                                                    range: 78..79,
                                                },
                                            ),
                                        ),
                                    ],
                                    values: [
                                        Literal(
                                            LiteralExpr {
                                                value: Int(
                                                    "5",
                                                ),
                                                range: 81..82,
                                            },
                                        ),
                                    ],
                                    range: 77..83,
                                },
                            ),
                            range: 71..83,
                        },
                    ),
                    range: 71..83,
                },
            ),
            Expression(
                ExpressionStmt {
                    value: Tuple(
                        TupleExpr {
                            elements: [
                                Await(
                                    AwaitExpr {
                                        value: Literal(
                                            LiteralExpr {
                                                value: Int(
                                                    "7",
                                                ),
                                                range: 90..91,
                                            },
                                        ),
                                        range: 84..91,
                                    },
                                ),
                                Literal(
                                    LiteralExpr {
                                        value: Int(
                                            "8",
                                        ),
                                        range: 93..94,
                                    },
                                ),
                            ],
                            range: 84..94,
                        },
                    ),
                    range: 84..94,
                },
            ),
            Expression(
                ExpressionStmt {
                    value: Await(
                        AwaitExpr {
                            value: Tuple(
                                TupleExpr {
                                    elements: [
                                        Literal(
                                            LiteralExpr {
                                                value: Int(
                                                    "9",
                                                ),
                                                range: 102..103,
                                            },
                                        ),
                                        Literal(
                                            LiteralExpr {
                                                value: Int(
                                                    "10",
                                                ),
                                                range: 105..107,
                                            },
                                        ),
                                    ],
                                    range: 101..108,
                                },
                            ),
                            range: 95..108,
                        },
                    ),
                    range: 95..108,
                },
            ),
            Expression(
                ExpressionStmt {
                    value: Await(
                        AwaitExpr {
                            value: Compare(
                                CompareExpr {
                                    left: Literal(
                                        LiteralExpr {
                                            value: Int(
                                                "1",
                                            ),
                                            range: 115..116,
                                        },
                                    ),
                                    ops: [
                                        Eq,
                                    ],
                                    comparators: [
                                        Literal(
                                            LiteralExpr {
                                                value: Int(
                                                    "1",
                                                ),
                                                range: 120..121,
                                            },
                                        ),
                                    ],
                                    range: 115..121,
                                },
                            ),
                            range: 109..121,
                        },
                    ),
                    range: 109..121,
                },
            ),
        ],
        range: 0..121,
    },
    parse_errors: [],
}
